src/validation.cpp

```diff
@@ -1070,6 +1070,52 @@ bool ReadBlockFromDisk(CBlock& block, const CBlockIndex* pindex, const Consensus
     return true;
 }
 
+bool ReadRawBlockFromDisk(std::vector<uint8_t>& block, const CDiskBlockPos& pos, const CMessageHeader::MessageStartChars& message_start)
+{
+    CDiskBlockPos hpos = pos;
+    hpos.nPos -= 8; // Seek back 8 bytes for meta header
+    CAutoFile filein(OpenBlockFile(hpos, true), SER_DISK, CLIENT_VERSION);
+    if (filein.IsNull()) {
+        return error("%s: OpenBlockFile failed for %s", __func__, pos.ToString());
+    }
+
+    try {
+        CMessageHeader::MessageStartChars blk_start;
+        unsigned int blk_size;
+
+        filein >> blk_start >> blk_size;
```

Comment:
compilation fails here, not sure if I'm missing a previous backport or what. 
Unable to call Unserialize for type obj

Reply:
`ReadRawBlockFromDisk` introduced here is only used for `MSG_WITNESS_BLOCK` which we don't have. I think it safe to do a partial backport and drop `ReadRawBlockFromDisk` functions.